### ISSUE TITLE
Keyword name should be capitalized is too strict

### ISSUE THREAD

**Issue by bhirsz** [bhirsz/robotframework-robocop#150]
Keyword name should be capitalized rule is being reported also on names with embedded arguments:
```
Keyword Name ${var} And So On
This Is Embedded ${Var} Argument
```
I think it would be best to ignore embedded arguments in name (even if it's not capitalized). What do you think? @mnojek 

This rule will also report on:
```
Get Data With ID
Login To UI With User
```
What about ignoring words that are all uppercase?

I believe we also miss rule for _ in name, it could be included with Keyword name checker (together with reserved words). 
```
This_Should_Not_Happen
Because Is Looks Better
```

**Comment by bhirsz** [comment]
We also need to test it with number, ie 
```
Get 3 Accounts
```
Would it pass or it will be reported since 3 is not uppercase?

**Comment by bhirsz** [comment]
I've also noticed that this rule reports on parsing errors (since most syntax issues are being parsed as keywords):
```
For  ${i}  IN  ${list}
    Log  ${i}
END
```
It will report on For and END, since it's invalid loop. 
Also here
```
My Keyword
...  ${arg}
... ${not_enough_spaces_after_eol_mark}
```

Those two problems will be covered by reserved words rules.

**Comment by bhirsz** [comment]
> We also need to test it with number, ie
> 
> ```
> Get 3 Accounts
> ```
> 
> Would it pass or it will be reported since 3 is not uppercase?

It will not report for numbers
